dig.go

```diff
@@ -35,8 +35,9 @@ import (
 )
 
 const (
-	_optionalTag = "optional"
-	_nameTag     = "name"
+	_optionalTag        = "optional"
+	_nameTag            = "name"
+	_allowUnexportedTag = "allowUnexported"
```

Comment:
I'm not 100% sold on this tag name.

1. I'm not sure about just "unexported," "allow unexported," or "ignore unexported"
2. I don't think we use camelCase for struct tags usually. [go-flags](https://pkg.go.dev/github.com/jessevdk/go-flags) uses kebab-case so "allow-unexported" or "ignore-unexported" might work. Just "unexported" obviously sidesteps this issue entirely.

I'm personally leaning towards either of the following but I welcome more suggestions.

- `ignore-unexported:"true"`, or
- `unexported:"ignore"` defaulting to `unexported:"error"` (this possibly allows us to add a different behavior for unexported fields in the future and use that name here, e.g. `unexported:"warn"` but I'm just spitballing)

Reply:
Following offline discussion, I've decided to go with `ignore-unexported:"true"`.